doc/developer/feature-lifecycle.md

```diff
@@ -0,0 +1,111 @@
+
+## Internal Development
+
+The goal of Internal Development is to build enough of a solution
+to a problem that we can demo it in front of users, internal and external,
+to test it. Depending on the size and scope of a project,
+Internal Development can take anywhere from a few days to a few months to complete.
+
+It is best practice for the DRI to break the solution into meaningful
+and manageable work items before anyone begins implementing it.
+Smaller, scoped work items provide good check in points for the
+DRI to communicate progress on the solution more widely.
+
+_Note: nothing shown or built in the Internal Development phase is
+expected to be stable, nor should it ever be available in a customer environment._
+
+Before a feature can move to Private Preview, it must:
+
+- Be code complete, unless we are intentionally putting a
+  partial solution in front of testers.
+- Have Stash migrations written and reviewed, if the code contains
+  breaking syntax changes.
+- Be stable enough that it will not:
+    - Impact other customer environments when ungated for a single customer.
+    - Impact other clusters/workloads, when ungated in a customer environment.
+    - Cause stability issues for other features.
+- Live behind a gated feature flag.
+- Have basic tests live and running.
+- Have basic reference documentation live behind a {{< private-preview >}} flag.
+
+## Private Preview
+
+The goal of Private Preview is to identify and address all of
+the reasons why the implementation of a feature might not work.
+To do this, the DRI is responsible for getting the feature in
+front of internal and external testers by posting in the
+#epd-announce channel.
+
+Testers may raise a few different classes of issues:
+[value, usability, feasibility, and viability](https://www.svpg.com/four-big-risks/).
+Some of these issues may be significant enough for the DRI to
+reconsider the chosen solution, effectively sending a project
+back to the Discovery phase. Other issues may be smaller, and
+fixes can be made within the scoped solution. The Private Preview
+phase may take a few days for smaller features, and up to a few
+weeks for larger ones.
+
+_Note: features in Private Preview will not be available in
+customer environments by default, they will be turned on for
+individual customers via the
+[limit increase process](https://www.notion.so/Adjusting-environment-limits-ba79e95871734ab2a292ecc8698d01e9).
+Customers who test these features should be warned that they
+are subject to significant, and occasionally incompatible, changes._
+
+Before a feature can move to Public Preview, it must:
+
+- Be verified that it solves the targeted problem for at least
+  one external tester, with additional optional internal testers.
+- Be entirely code complete, if it wasn’t already.
+- Have comprehensive tests live and running.
+- Have basic observability live and running.
+- Have basic metrics live and running.
```

Comment:
Should projects like `terraform`, `dbt`, and `mz` include metrics and observability? I am inclined towards collecting metrics and enabling observation (e.g. Sentry), although there might be cases where it's not feasible or excessive.

Reply:
I'll resolve this comment, but please raise this with @morsapaes if it is unclear. She can drive the resolution forward on this.